devtools/startup/DevToolsStartup.jsm

```diff
@@ -1899,6 +1904,16 @@ function onRecordingStarted(recording) {
     reloadAndClearRecordingState(browser);
 
     recordReplayLog(`FinishedRecording ${recordingId}`);
+  });
+
+  recording.on("saved", function(name, data) {
+    const recordingId = data.id;
+
+    if (!browser) {
+      // Be sure we received the finished event before the saved event
+      handleFailedRecording(browser, 'Recording processed out of order');
+      return;
+    }
```

Comment:
I think we should do this in `finished` instead. It should be fine to assume that if `saved` ran then `finished` must have run first.

Reply:
I don't think we need this in `finished` (at least in this form) because `getBrowser()` will `throw` if it can't find the browser so it wouldn't reach this block. It exists in `saved` primarily for completeness in case a refactor happens that introduces a condition in which `browser` is undefined. In the current flow, this can't happen.

Reply:
Ahh got it. As is I realize this code won't work anyway because we're guaranteed to be doing calling `handleFailedRecording(null`, so it will just throw an exception anyway. I'd vote to delete this.

Reply:
Fair point!